src/common/entities/assignments.entity.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  Entity,
+  EntityRepositoryType,
+  ManyToOne,
+  PrimaryKey,
+  Property,
+  Rel,
+} from "@mikro-orm/core";
+
+import { AssignmentsRepository } from "@/assignments/assignments.repository";
+
+import { Classroom } from "./classrooms.entity";
+import { CustomBaseEntity } from "./custom-base.entity";
+
+@Entity({ tableName: "assignment", repository: () => AssignmentsRepository })
+export class Assignment extends CustomBaseEntity {
+  [EntityRepositoryType]?: AssignmentsRepository;
+
+  @PrimaryKey()
+  id!: number;
+
+  @Property({ fieldName: "title" })
+  title!: string;
+
+  @Property({ fieldName: "description" })
+  description!: string;
+
+  @Property({ fieldName: "due_date" })
+  dueDate!: Date;
+
+  @ManyToOne(() => Classroom, { fieldName: "classroomId" })
```

Comment:
```suggestion
  @ManyToOne(() => Classroom, { fieldName: "classroom_id" })
```

Db field names should be snake_case